test/extensions/filters/http/admission_control/admission_control_filter_test.cc

```diff
@@ -0,0 +1,249 @@
+#include <chrono>
+
+#include "envoy/extensions/filters/http/admission_control/v3alpha/admission_control.pb.h"
+#include "envoy/extensions/filters/http/admission_control/v3alpha/admission_control.pb.validate.h"
+
+#include "common/stats/isolated_store_impl.h"
+
+#include "extensions/filters/http/admission_control/admission_control.h"
+#include "extensions/filters/http/admission_control/evaluators/response_evaluator.h"
+#include "extensions/filters/http/admission_control/thread_local_controller.h"
+
+#include "test/mocks/runtime/mocks.h"
+#include "test/mocks/server/mocks.h"
+#include "test/mocks/thread_local/mocks.h"
+#include "test/test_common/simulated_time_system.h"
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::NiceMock;
+using testing::Return;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace AdmissionControl {
+namespace {
+
+class MockThreadLocalController : public ThreadLocal::ThreadLocalObject,
+                                  public ThreadLocalController {
+public:
+  MockThreadLocalController() = default;
+  MOCK_METHOD(uint32_t, requestTotalCount, (), (override));
+  MOCK_METHOD(uint32_t, requestSuccessCount, (), (override));
+  MOCK_METHOD(void, recordSuccess, (), (override));
+  MOCK_METHOD(void, recordFailure, (), (override));
+};
+
+class MockResponseEvaluator : public ResponseEvaluator {
+public:
+  MockResponseEvaluator() = default;
+  MOCK_METHOD(bool, isHttpSuccess, (uint64_t code), (const, override));
+  MOCK_METHOD(bool, isGrpcSuccess, (uint32_t status), (const, override));
+};
+
+class TestConfig : public AdmissionControlFilterConfig {
+public:
+  TestConfig(const AdmissionControlProto& proto_config, Runtime::Loader& runtime,
+             TimeSource& time_source, Runtime::RandomGenerator& random, Stats::Scope& scope,
+             ThreadLocal::SlotPtr&& tls, MockThreadLocalController& controller,
+             std::shared_ptr<ResponseEvaluator> evaluator)
+      : AdmissionControlFilterConfig(proto_config, runtime, time_source, random, scope,
+                                     std::move(tls), evaluator),
```

Comment:
`std::move` the evaluator

Reply:
clang-tidy used to complain about this particular gaffe. Not sure why that changed.. thanks